src/scss/components/_menu.scss

```diff
@@ -24,14 +27,43 @@ $menu-link-padding          : spacing() spacing(x2) !default;
         @include font-size($menu-fontSize);
     }
 
+        // Reduces the spacing between menu items
+        .c-menu--condensed {
+            .c-menu-link {
+                padding-bottom: $menu--condensed-link-padding;
+                padding-top: $menu--condensed-link-padding;
+            }
+        }
+
+        .c-menu--horizontal {
```

Comment:
Would be good to have a one-line comment saying (roughly) what this modifier does